torch/lib/THD/test/command_channel_smoke.cpp

```diff
@@ -24,26 +24,18 @@ void init_worker(const int& rank, const std::string& master_addr) {
 
   assert(channel->init());
 
-  // Send.
-  rpc::ByteArray arr;
-  arr.append("hello to master from worker ",
-      sizeof("hello to master from worker ") - 1);
-  arr.append(std::to_string(rank).c_str(), std::to_string(rank).size());
-
-  fprintf(stderr, "worker %d: about to send a message\n", rank);
-
-  auto rpc_msg = std::unique_ptr<rpc::RPCMessage>(new rpc::RPCMessage(std::move(arr)));
-  channel->sendMessage(std::move(rpc_msg));
-
-  fprintf(stderr, "worker %d: sent message\n", rank);
-
-  // Recieve.
   auto msg = channel->recvMessage();
   std::string expected = std::string("hello to worker ") +
       std::to_string(rank) + " from master";
   fprintf(stderr, "Worker %d: received '%.*s'\n", rank,
       (int)msg.get()->bytes().length(), msg.get()->bytes().data());
   assert(expected.compare(msg.get()->bytes().to_string()) == 0);
+
+  /*
+   * We need to sleep for a while because instant exiting will cause closing
+   * the socket and in result error in `errorHandler` function.
```

Comment:
Oh, maybe its not well described. But yes, we have to wait until master will do all the receiving/sending and only then we can exit. It is because when worker exit it will close socket and trigger poll in error thread what then causes exception.

I think, I have an idea to change `sleep` to `conditional_variable` and `mutex` to be sure that we can safely exit instead of relying on sleeping for 'magic' time.